evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -82,17 +85,19 @@ def _get_cat_cols(X):
 
     def fit(self, X, y=None):
         top_n = self.parameters['top_n']
-        if not isinstance(X, pd.DataFrame):
-            X = pd.DataFrame(X)
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
         X_t = X
-
         if self.features_to_encode is None:
             self.features_to_encode = self._get_cat_cols(X_t)
         invalid_features = [col for col in self.features_to_encode if col not in list(X.columns)]
         if len(invalid_features) > 0:
             raise ValueError("Could not find and encode {} in input data.".format(', '.join(invalid_features)))
-
         if self.parameters['handle_missing'] == "as_category":
+            for col in self.features_to_encode:
+                if X_t[col].dtype == 'category' and pd.isna(X_t[col]).any():
+                    X_t[col] = X_t[col].cat.add_categories("nan")
+                    X_t[col] = X_t[col].where(pd.isna(X_t[col]).isin([False]), other='nan')
```

Comment:
So this says that we'll fill anything for which `pd.isna` is true (i.e. `pd.NA`, `np.nan`, `None` I believe?) with `np.nan`?

I see we use this code in `fit` and in `transform`. Is there some way we can move this for loop code to a common private helper method, to avoid duplication? Or would that be overkill in this instance?